src/sql/workbench/contrib/scripting/browser/scriptingActions.ts

```diff
@@ -307,39 +310,35 @@ CommandsRegistry.registerCommand({
 // Refresh Action for Scriptable objects
 CommandsRegistry.registerCommand({
 	id: OE_REFRESH_COMMAND_ID,
-	handler: async (accessor, args: ObjectExplorerActionsContext) => {
-		const connectionManagementService = accessor.get(IConnectionManagementService);
+	handler: async (accessor, args: ObjectExplorerActionsContext): Promise<void> => {
 		const capabilitiesService = accessor.get(ICapabilitiesService);
 		const objectExplorerService = accessor.get(IObjectExplorerService);
-		const errorMessageService = accessor.get(IErrorMessageService);
+		const logService = accessor.get(ILogService);
+		const notificationService = accessor.get(INotificationService);
 		const connection = new ConnectionProfile(capabilitiesService, args.connectionProfile);
-		let treeNode: TreeNode;
 		if (connectionManagementService.isConnected(undefined, connection)) {
-			treeNode = await getTreeNode(args, objectExplorerService);
-			if (treeNode === undefined) {
-				objectExplorerService.updateObjectExplorerNodes(connection.toIConnectionProfile()).then(() => {
-					treeNode = objectExplorerService.getObjectExplorerNode(connection);
-				});
+			let treeNode = await getTreeNode(args, objectExplorerService);
+			if (!treeNode) {
+				await objectExplorerService.updateObjectExplorerNodes(connection.toIConnectionProfile());
+				treeNode = objectExplorerService.getObjectExplorerNode(connection);
+			}
+			if (treeNode) {
+				const tree = objectExplorerService.getServerTreeView().tree;
+				try {
+					await tree.collapse(treeNode);
+					await objectExplorerService.refreshTreeNode(treeNode.getSession(), treeNode);
+					await tree.refresh(treeNode);
+					await tree.expand(treeNode);
+				} catch (err) {
+					// Display message to the user but also log the entire error to the console for the stack trace
+					notificationService.error(localize('refreshError', "An error occurred refreshing node '{0}': {1}", args.nodeInfo.label, getErrorMessage(err)));
```

Comment:
I'd love to know how often this happens to our users. Would we have any insights (even in an aggregated form) here, or would we need a bunch of infrastructure setup to do that?

If not, something to consider for the future. Would be great to understand what and how many errors are occurring most for users, so that we can drive down those occurrences over time.

Reply:
Agreed. I won't add that here but let me think about how best to surface errors like that here to our telemetry (hopefully in a generic way so we can get it for free as long as we have a standard way of logging errors)

Reply:
sounds good. thanks.

Reply:
vscode has code to add telemetry for unexpected errors, but we disabled it for pii reasons.